### PR TITLE
Add a flag for message queue availability (#73)

### PR DESCRIPTION
**Pull request by yairogen**

### CODE REVIEW

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java [hunk 1]
```diff
@@ -61,6 +62,10 @@
     static AtomicBoolean IS_CONNECTED = new AtomicBoolean(false);
```
Comment (nahsh) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: IS_CONNECTED - is it in use?

**Comment by yairogen** [comment]
We use it triggerReconnectThread

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java (hunk 1 above)
Comment (yairogen) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: We use it triggerReconnectThread

File: queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java [hunk 2]
```diff
@@ -61,6 +62,10 @@
...
     static CountDownLatch INIT_LATCH = new CountDownLatch(1);
     static AtomicBoolean IS_BLOCKED = new AtomicBoolean(false);
+    static AtomicBoolean IS_CONN_OR_CH_UP_INT = new AtomicBoolean(false);
```
Comment (igreenfield) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: I don't like the name....

Comment (yairogen) [inline queue-rabbitmq/src/main/java/com/cisco/oss/foundation/message/RabbitMQMessagingFactory.java]: and now?

**Pull request merged**